src/ipelib/ipegeo.cpp

```diff
@@ -844,6 +844,27 @@ Rect Bezier::bbox() const {
     return Rect(box.bottomLeft() - Vector(0.5, 0.5), box.topRight() + Vector(0.5, 0.5));
 }
 
+Vector Arc::midpoint() const {
+    // Normalize iBeta with respect to iAlpha to handle cases where iAlpha > iBeta
+    Angle betaNorm = iBeta;
+    betaNorm.normalize(iAlpha);
+    double delta = betaNorm - iAlpha;
+
+    if (sq(delta - IpeTwoPi) < 1e-20) {
```

Comment:
When the arc is an entire ellipse, iBeta = iAlpha + 2 pi, and then betaNorm will be normalized to iAlpha, so that delta is zero.   If we want to cover this case, we should explicitly check for `iBeta - iAlpha > IpeTwoPi - 1e-10` and for `iAlpha - 1e-10 < iBeta < iAlpha`.  (There is a method `isEllipse`, but that will trigger only for explicitly constructed full ellipses, not when the user has just drawn one.) 

I'm not sure it is actually possible to draw an arc that will trigger this, as when you let the end point snap to the start point, Ipe will interpret that as a zero-length arc, not a 360 degree one.   So my feeling is that we may not need to add special handling for this case.

Reply:
Thank you for the review! I think it makes sense to cover the case, as it could potentially be triggered when e.g. reading XML or when future code calls the method.